File: examples/lua53/client/protocols/ReqSceneEnterFly.lua
local M = {}
local moduleName = ...
_G[moduleName] = M
setmetatable(M, {__index = _G})
local _ENV = M


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = ReqSceneEnterFly})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.map_id = 0
end


function encode(self)
	pack = packet.new()
	pack:writeU32(self.map_id)
	return pack:encode(Msg.P_REQ_SCENE_ENTER_FLY)
end


function setMapId(self, map_id)
	self.map_id = map_id
end

File: data/data.code.typescript.cc.client/ReqSceneEnterFly.ts
import { Packet } from '@mi/mod/Packet'


export class ReqSceneEnterFly
{
	private _map_id: number;


	public Encode(): Packet {
		let packet: Packet = new Packet();
		packet.WriteUint(this._map_id);
		packet.Encode(2010);
		return packet;
	}


	public get map_id(): number { return this._map_id; }
	public set map_id(value: number) { this._map_id = value; }
}
